### PR TITLE
Refactor of Standard Deployment to AZD

### PR DESCRIPTION
**Pull request by hatboyzero**
# Refactor of Standard Deployment to AZD

## Details on the issue fix or feature implementation

Initial pass at refactoring Standard Deployment to AZD

## Confirm the following

- [ ]  I started this PR by branching from the head of the default branch
- [ ]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [ ]  I have included inline docs for my changes, where applicable
- [ ]  I have successfully run a local build
- [ ]  I have provided the required update scripts, where applicable
- [ ]  I have updated relevant docs, where applicable

### CODE REVIEW

File: deploy/standard/infra/README.md [hunk 1]
```diff
@@ -1,90 +1,90 @@
...
-### Additional Notes:
-- **Azure OpenAI Service Regions:**
-  - Ensure that the `<location>` value in your deployment script corresponds to a region that supports Azure OpenAI services. Refer to Azure OpenAI service regions for more information.
-
-## Post-deployment configuration
-
-### Authentication setup
-
-Follow the instructions in the [Authentication setup document](https://docs.foundationallm.ai/deployment/authentication/index.html) to configure authentication for the solution.
+# Standard Deployment with AKS (bicep)
```
Comment (jamesrcounts) [inline deploy/standard/infra/README.md:1]: The contents of this file should be truncated and replaced with a reference to the docs site.

File: deploy/standard/infra/bootstrap.ps1
Comment (jamesrcounts) [inline deploy/standard/infra/bootstrap.ps1:1]: As this is related to psake, we can remove it.  Does it do anything else?

File: deploy/standard/infra/main.bicep [hunk 2]
```diff
@@ -211,7 +211,7 @@ module storage 'storage-rg.bicep' = {
 }
 
 module vec 'vec-rg.bicep' = {
-  dependsOn: [rg]
+  dependsOn: [rg, networking]
```
Comment (jamesrcounts) [inline deploy/standard/infra/main.bicep:214]: are these explicit depends left over from testing `azd down`?  Can we remove them (excluding `rg`)?

**Review by jamesrcounts — CHANGES_REQUESTED** [review]

**Comment by jamesrcounts** [comment]
In addition to the comments in the review, delete `psakefile.ps1`

File: deploy/standard/azd-hooks/postprovision.ps1 [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            -resourceGroup $resourceGroup `
+            -subscription $env:AZURE_SUBSCRIPTION_ID
+    }
+}
+finally {
+    Pop-Location
+    Set-PSDebug -Trace 0 # Echo every command (0 to disable, 1 to enable)
+}
+
+# function envsubst {
```
[the remark this answers is not in the document]
Comment (hatboyzero) [reply deploy/standard/azd-hooks/postprovision.ps1:44]: No, those comments need to be removed.

File: deploy/standard/infra/bootstrap.ps1
Comment (hatboyzero) [reply deploy/standard/infra/bootstrap.ps1:1]: No, it only bootstraps Psake.  I will remove it.

File: deploy/standard/infra/main.bicep (hunk 2 above)
Comment (hatboyzero) [reply deploy/standard/infra/main.bicep:214]: Yes, they can be removed.

File: deploy/standard/azd-hooks/postprovision.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (hatboyzero) [reply deploy/standard/azd-hooks/postprovision.ps1]: Comments are removed.

File: deploy/standard/azd-hooks/utility/Upload-AuthStoreData.ps1 [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        --output tsv
+}
+
+if (-not (Test-Path "../data/role-assignments/$($instanceId)`.json")) {
+    throw "Default role assignments json not found at ../data/role-assignments/$($instanceId)`.json"
+}
+
+$target = "https://$storageAccountAdls.blob.core.windows.net/role-assignments/"
+
+& ../../common/tools/azcopy/azcopy cp "../data/role-assignments/$($instanceId)`.json" "$target"
```
[the remark this answers is not in the document]
Comment (hatboyzero) [reply deploy/standard/azd-hooks/utility/Upload-AuthStoreData.ps1]: Done

File: deploy/standard/azd-hooks/utility/Upload-SystemPrompts.ps1 [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+$storageAccount = Invoke-AndRequireSuccess "Getting storage account name" {
+    az storage account list `
+        --resource-group $resourceGroup `
+        --query '[0].name' `
+        --output tsv
+}
+
+$target = "https://$storageAccount.blob.core.windows.net/resource-provider/"
+
+& ../../common/tools/azcopy/azcopy copy '../../common/data/resource-provider/*' $target `
```
[the remark this answers is not in the document]
Comment (hatboyzero) [reply deploy/standard/azd-hooks/utility/Upload-SystemPrompts.ps1]: Done

File: deploy/standard/infra/bootstrap.ps1
Comment (hatboyzero) [reply deploy/standard/infra/bootstrap.ps1:1]: Removed bootstrap.ps1 and Psakefile.ps1

File: deploy/standard/infra/main.bicep (hunk 2 above)
Comment (hatboyzero) [reply deploy/standard/infra/main.bicep:214]: Actually, we need to leave these.  While it doesn't affect teardown, it does affect provisioning.  Updating them to only reflect the necessary dependencies.

File: deploy/standard/infra/README.md (hunk 1 above)
Comment (hatboyzero) [reply deploy/standard/infra/README.md]: Updated to point to online docs and the top level index of setup-guides for now.  Once official standard deployment docs are added to the setup-guides, we can update this to point directly to that then.

File: deploy/standard/azd-hooks/utility/Upload-AuthStoreData.ps1 [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+$storageAccountAdls = Invoke-AndRequireSuccess "Get ADLS Auth Storage Account" {
+    az storage account list `
+        --resource-group $resourceGroup `
+        --query "[?kind=='StorageV2'].name | [0]" `
+        --output tsv
+}
+
+$sourceJson = "../data/role-assignments/$($instanceId)`.json" | Resolve-Path
+
+if (-not (Test-Path $sourceJson)) {
```
Comment (jamesrcounts) [inline deploy/standard/azd-hooks/utility/Upload-AuthStoreData.ps1:56]: The resolve path cmdlet will blow up if the path is not found.

File: deploy/standard/infra/main.bicep [hunk 7]
```diff
@@ -3,28 +3,30 @@ targetScope = 'subscription'
...
-@secure()
-param authClientSecret string
+param userPortalHostname string
+param managementPortalHostname string
+param coreApiHostname string
+param managementApiHostname string
 
 // Locals
 var abbrs = loadJsonContent('./abbreviations.json')
+// var authAppRegistrationScopes = 'api://FoundationaLLM-Authorization-Auth/.default'
```
Comment (jamesrcounts) [inline deploy/standard/infra/main.bicep:28]: Do we need this comment?

**Review by jamesrcounts — CHANGES_REQUESTED** [review]

File: deploy/standard/infra/main.bicep (hunk 7 above)
Comment (hatboyzero) [reply deploy/standard/infra/main.bicep:28]: No, I will remove it.

Comment (hatboyzero) [reply deploy/standard/infra/main.bicep]: Removed

File: deploy/standard/azd-hooks/utility/Upload-AuthStoreData.ps1 (hunk 6 above)
Comment (hatboyzero) [reply deploy/standard/azd-hooks/utility/Upload-AuthStoreData.ps1:56]: Updated

**Review by jamesrcounts — APPROVED** [review]

**Review by ciprianjichici — APPROVED** [review]

**Review by ciprianjichici — APPROVED** [review]

**Pull request merged**